heron/scheduler-core/tests/java/com/twitter/heron/scheduler/RuntimeManagerMainTest.java

```diff
@@ -204,6 +209,35 @@ public void testManageTopologyFailCall() throws Exception {
   }
 
   @PrepareForTest(ReflectionUtils.class)
+  @Test(expected = UpdateDryRunResponse.class)
+  public void testManageTopologyDryRun() throws Exception {
+    config = mock(Config.class);
+    when(config.getStringValue(ConfigKeys.get("TOPOLOGY_NAME"))).thenReturn(TOPOLOGY_NAME);
+
+    RuntimeManagerMain runtimeManagerMain = spy(new RuntimeManagerMain(config, MOCK_COMMAND));
+    // Valid state manager class
+    Mockito.when(config.getStringValue(ConfigKeys.get("STATE_MANAGER_CLASS"))).
+        thenReturn(IStateManager.class.getName());
+    PowerMockito.mockStatic(ReflectionUtils.class);
+    PowerMockito.doReturn(Mockito.mock(IStateManager.class))
+        .when(ReflectionUtils.class, "newInstance", Mockito.eq(IStateManager.class.getName()));
+
+    // Legal request
+    UpdateDryRunResponse mockResponse = new UpdateDryRunResponse(
+        TopologyAPI.Topology.getDefaultInstance(), config, mock(PackingPlan.class),
+        mock(PackingPlan.class), new HashMap<String, Integer>());
+    doNothing().when(runtimeManagerMain)
+        .validateRuntimeManage(any(SchedulerStateManagerAdaptor.class), eq(TOPOLOGY_NAME));
+    // Successfully get ISchedulerClient
+    ISchedulerClient client = mock(ISchedulerClient.class);
+    doReturn(client).when(runtimeManagerMain).getSchedulerClient(any(Config.class));
+    // Happy path
+    doThrow(mockResponse).when(runtimeManagerMain).
```

Comment:
I've been trying to do a really mock test here. This is what I have done so far: https://github.com/twitter/heron/pull/1629/files#diff-581c3ba72d069451257c5910ad628e4eR216

But as discussed, I can't really work out a correct unit test because I keep encountering NPE. [See the log](https://gist.github.com/objmagic/9cd19aa781fc39ca8580e4b93e4365cf). I was following your example at [RuntimeManagerRunnerTest](https://github.com/twitter/heron/blob/master/heron/scheduler-core/tests/java/com/twitter/heron/scheduler/RuntimeManagerRunnerTest.java#L219) but it seems that it did not work in my case...